.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,36 @@
+name: Publish Docs
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - "docs/**"
+      - ".github/workflows/publish-docs.yml"
+      - "Makefile"
+
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: |
+        make st2
+        sudo apt-get update
+        # python3-dev is already available
+        # sudo apt install python3-dev
+        sudo apt install libldap2-dev 
+        sudo apt install libsasl2-dev
+        sudo pip3 install virtualenv
+        make docs
```

Comment:
Good start, 

Per existing CircleCI config, there's also a need for additional routing logic depending on the branch:
https://github.com/StackStorm/st2docs/blob/6c65852f42ea6a687c1b87e25ad67a320baa1538/.circleci/config.yml#L41-L61

From what I understand the logic looks this way:
* `master` branch build -> `/latest/` st2docs dir
* `v.X.Y.Z` latest stable branch -> `/` st2docs dir
* other `vX.Y.Z` branch build -> `vX.Y` st2docs dir

Reply:
this might be possible with the `destination_dir: subdir` option

Reply:
so, the `destination_dir: subdir` option works well for `/latest` and `/X.Y` site directories.  so 🎉 

One difference, however, is that if we deploy to the root for the latest GA version like we do for S3, we'll lose all other subdirectories.  For example, if we have `/1.0` and `/latest`, and then we determine that the current version is also the latest GA release, deploying the site to `/` will wipe away `/1.0` and `/latest`.  I've confirmed this with testing.

To get around this, we could implement the action to deploy the latest GA release to a `/ga` subdirectory, and we can use URL routing to point `docs.stackstorm.com` to `docs.stackstorm.com/ga`.  Thoughts?

Reply:
actually, we have a bigger issue:

When we push to github pages, a bot will run a `pages-build-deployment` pipeline.  If we push to `gh-pages` again while an existing `pages-build-deployment` pipeline is running, github will cancel the previous deployment in favor of the subsequent one:
![image](https://user-images.githubusercontent.com/1030325/174670241-96dda775-91d8-4399-a339-cefe69516e82.png)

This means that we cannot push to `gh-pages` while there is an ongoing gh pages deployment, and it is non-trivial to "wait" until the gh pages deployment is finished.  I suspect it's possible, though I've not looked into how it might be done (it'd probably involve the github api and some sort of polling logic).

This will also impact us when we want to run a deployment to existing version branches as we'd have to wait until each is finished before triggering the next one..

One thing that might help is that we could utilize the concurrency feature for our workflow.  This will help ensure that our workflows don't run while another of our workflows are running, but we don't seem to have control over the `pages-build-deployment` pipeline.
https://stackoverflow.com/questions/70631168/github-actions-going-through-with-pages-build-deployment

Using `concurrency` might _help_ because it's likely the gh pages will finish before any subsequent builds do, but it's a risk.  It's possible that the gh pages deployment will take some time in which case our st2docs build _might_ finish before and cancel the existing `pages-build-deployment` pipeline.
Concurrency was not working with my testing of multiple jobs in the same workflow, however.. (test here: https://github.com/StackStorm/st2docs/pull/1129/files#diff-7b112bd01a3c7000d73727a7e66cbfb9cf00a47702c94fda157e23f46e20db17R1)

However, it also means that we have to run two builds for the same thing...

One possible workaround is to use routing logic in Route53 or cloudfront to point the latest GA release instead of using a `/ga` subdirectory.  I can think of 2 ways to do this:
1. have some logic in the github action pipeline that modifies entries in route53/cloudfront
2. manually update entries in route53/cloudfront as part of the GA release process

alternatively, it _might_ be possible in the future to publish to GH Pages without the bot, but for now, that's our only option, I think:
https://github.community/t/github-pages-bot-added-to-our-repos-unannounced/218214/13

Reply:
example pipeline here: https://github.com/mickmcgrath13/st2docs/actions/runs/2531438591

Reply:
i think the issue was that if we are deploying say to version 3.7 of the docs, and then do a deploy to latest whilst its still running the deploy for v3.7. The v3.7 one will be cancelled, and we won't get the update to v3.7.

Reply:
But the time `pages-build-deployment` runs, the updated 3.7 sources will already be committed to the gh-pages branch. `pages-build-deployment` redeploys everything in `gh-pages` afaik, so the next `pages-build-deployment` run will include both sets of changes.

Reply:
from my [previous notes](https://github.com/StackStorm/st2docs/pull/1129#discussion_r901967475):
> One difference, however, is that if we deploy to the root for the latest GA version like we do for S3, we'll lose all other subdirectories. For example, if we have /1.0 and /latest, and then we determine that the current version is also the latest GA release, deploying the site to / will wipe away /1.0 and /latest. I've confirmed this with testing.

One thing that can possibly work around these issues is to always only build gh pages in a pipeline based on the base branch.  the base branch gh-pages pipeline will also `clone` other versions/subdirectories and build those.

then we have pipelines for version-specific branches that just trigger the base gh-pages pipeline to rebuild everything

an all-in-one pipeline at the base branch that builds root and subdirectories would also mitigate special routing requirements (i.e. we just build all the subdirectories we need)

just some thoughts..